include/pisa/util/inverted_index_utils.hpp

```diff
@@ -32,9 +32,16 @@ void sample_inverted_index(std::string const &input_basename,
     auto document_count = static_cast<uint32_t>(input.num_docs());
     write_sequence(dos, gsl::make_span<uint32_t const>(&document_count, 1));
     pisa::progress progress("Sampling inverted index", input.size());
+    size_t term = 0;
     for (auto const &plist : input) {
-        auto sample = sample_fn(plist.docs.size());
-        assert(std::is_sorted(std::begin(sample),std::end(sample)));
+        auto sample = sample_fn(plist.docs);
+        if (sample.size() == 0) {
+            dropped_term_ids.push_back(term);
```

Comment:
`dropped_term_ids.push_back(term++)?`

Reply:
Actually, at this rate, why don't we just say:
```cpp
for (auto term = 0; term < input.size(); term += 1, progress.update(1)) {
    auto& plist = input[term];
    ...
}
```
Less error-prone than keeping track of the counters in two places.

Reply:
We cannot do the above @elshize, there is no `operator[]` for `binary_freq_collection` for now. We need to add it, but I dont want to do it in this PR as it might break other things.